File: src/Web/control-plane-ui/src/models/camera.ts
import { SIM } from "./sim";

export enum CameraType {
    FiveG = "5G",
    FourG = "4G",
    Wifi = "Wi-fi",
    Ethernet = "Ethernet",
    LTE = "LTE",
    Container = "Container"
}

export type Camera = {
    id: string;
    model: string;
    type: CameraType;
    ip?: string;
    username?: string;
    password?: string;
    port?: string;
    simId?: string;
    rtsp?: string;
    hls?: string;
}

File: src/Services/ControlPlane/API/Models/Camera.cs
namespace ControlPlane.API.Models
{
    public record Camera(
        string Id,
        string? Model,
        string? Type,
        string? Ip,
        string? Port,
        string? SimId,
        string? Rtsp,
        string? Hls,
        string? Username,
        string? Password
    );
}